mne/time_frequency/psd.py

```diff
@@ -158,7 +163,12 @@ def compute_epochs_psd(epochs, picks=None, fmin=0, fmax=np.inf, n_fft=256,
                            exclude='bads')
     n_fft, n_overlap = _check_nfft(len(epochs.times), n_fft, n_overlap)
 
-    data = epochs.get_data()[:, picks]
+    if tmin is not None or tmax is not None:
+        time_mask = _time_mask(epochs.times, tmin, tmax)
```

Comment:
That's why it's `allow_slice` and not `return_slice`, because it will only return a slice if it's possible to do so. This is analogous to some `numpy` functions that return a view if possible. It should always be possible with `_time_mask`, but maybe possible only some of the time with `picks`, as you point out. We can make it clear in docs that this is the behavior. That's also why `allow_slice=False` would be the default, so that we don't break existing things that assume a copy will be made when indexing. Gradually we can go through and optimize other parts of the code that can tolerate potential views.